src/discovery/Discovery.ts

```diff
@@ -117,8 +117,7 @@ class Discovery {
         if (nodeId.equals(this.nodeManager.getNodeId())) {
           const vertexChainDataEncoded = await this.nodeManager.getChainData();
           // Decode all our claims - no need to verify (on our own sigchain)
-          for (const c in vertexChainDataEncoded) {
-            const claimId = c as ClaimIdEncoded;
+          for (const claimId in vertexChainDataEncoded) {
```

Comment:
This would be `ClaimIdString`, not `ClaimIdEncoded`. I'm going to add that type into `claims/types.ts` in case you want to use it.

Reply:
It's better to name this `claimIdString` instead of `claimId` to be clear.

Reply:
What I'm checking this now, and this is incorrect.

The `vertexChainDataEncoded` is acquired from `await this.nodeManager.getChainData()`.

The type of `getChainData` is:

```
type ChainDataEncoded = Record<ClaimIdEncoded, ClaimEncoded>;
```

Which would mean that the keys are `ClaimIdEncoded` and not `ClaimId`.

Therefore, they are encoded strings. Is that intentional? Or do we want them to be binary string form?

Reply:
Furthermore, `nodeManager.getChainData()` is just a proxy to `sigchain.getChainData()`. Let's avoid proxying calls in `NodeManager` that are unnecessary and just fetch it from `Sigchain`.

Same with `getNodeId()`.

Reply:
I think you mentioned that the `SigChain` should use the encoded form since that would likely be seen by users at some point?

I think you also stated that the sigchain should be proxied same ans the nodeGraph calls.

Reply:
@CMCDragonkai and I were discussing how sigchain's `getChainData` is a bit incorrect.

Currently, `getChainData` returns the `ChainDataEncoded` type: `type ChainDataEncoded = Record<ClaimIdEncoded, ClaimEncoded>;`

However, @CMCDragonkai pointed out that the `Record` type does not preserve order. This is an issue for the sigchain data, as in order to verify a claim, we need to compute the hash of the chronologically previous claim, and compare with the hash stored in the current claim.

(It slightly confuses me how my tests were passing with this then - I'd need to take a closer look at the tests to see why they'd be passing.)

Suggestions to improve this:
* create a `getChain` function that returns the claims on the sigchain as an **array** of `Claim` types (i.e. the unencoded POJO , retaining the chronological order of the sigchain. This would be expected to be used for all internal functions for the local keynode (i.e. not to be transferred across the wire). i.e. the following:
    ```
    [
      { payload: { ... }, signatures: { ... } },
      { payload: { ... }, signatures: { ... } },
      ...
    ]
    ```
* extend this with a `seek` and `limit` functionality to be able to acquire a specific claim (according to ID/sequence number), and limit the number of claims returned, respectively.
* create a `getChainEncoded` function that returns the claims on the sigchain as an **array** of `ClaimEncoded` types (i.e. the encoded base64 string version of the claim), but still retaining the chronological order of the sigchain. This would be used when transferring across the network, such that the receiving keynode can correctly verify the received sigchain by chronologically moving through it

Reply:
I was mistaken. this comment https://github.com/MatrixAI/js-polykey/pull/310#issuecomment-1025303139 states that `Sigchain` should be used directly.

Reply:
So `Sigchain` class API should be changed in a later PR, but not this PR. This PR I just needed to ensure the right keywords are being used during iteration, at first I thought it was `ClaimIdString` but actually it's `ClaimIdEncoded`. So it should be `for (const claimIdEncoded in vertexChainDataEncoded) { ... }` for now.

Reply:
Put this into a new issue in #327.

Reply:
Ok updated this to `claimIdEncoded`.

This is a temporary fix, because the whole discovery has been changed by @emmacasolin in a later PR anyway. And we are going to change the sigchain API anyway in #327.

Note that I also benchmarked `for .. in` vs `Object.entries`, and the former is about 50% faster than the latter in the chrome browser.